### PR TITLE
Charge correction via alchemical waters

### PR DESCRIPTION
**Pull request by IAlibay**
## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT License [here](https://github.com/OpenFreeEnergy/openfe/blob/main/LICENSE) and the **Developer Certificate of Origin** at <https://developercertificate.org/>.

### CODE REVIEW

**Comment by IAlibay** [comment]
no idea what's going on with codecov.. it's pointing to stuff not being covered that should

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py
Comment (richardjgowers) [inline openfe/protocols/openmm_rfe/equil_rfe_settings.py:101]: Can these be pulled automatically from the SolventComponent solvation settings?

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py
Comment (richardjgowers) [inline openfe/protocols/openmm_rfe/equil_rfe_settings.py:107]: this is probably not a real setting, isn't this just the resname that `add_solvent()` adds?  (i.e. it's know a priori and can be hard coded?)

File: openfe/protocols/openmm_rfe/equil_rfe_methods.py
Comment (richardjgowers) [inline openfe/protocols/openmm_rfe/equil_rfe_methods.py:79]: these first lines in docstrings are meant to be a one liner.  So `Difference in formal charge between state A and B` gives the same information

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py
Comment (IAlibay) [reply openfe/protocols/openmm_rfe/equil_rfe_settings.py:107]: Todo - I need to double check the FF files to see if the resname is inherited or set by modeller.


However! This does bring up an interesting thought, this won't work with 4+ site waters because we don't handle vsite transformations. I need to add in a guard & some docs for it.

File: openfe/protocols/openmm_rfe/equil_rfe_methods.py
Comment (IAlibay) [reply openfe/protocols/openmm_rfe/equil_rfe_methods.py]: done

**Comment by IAlibay** [comment]
@richardjgowers - so looks like we can assume the residue name _if_ we assume we're picking up auto-generated solvent.

I'll switch the code out for that, however in the longer term we should keep an eye out on if users might want to manually define an ion / water in their PDB that they want to be the alchemical water.

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py
Comment (IAlibay) [reply openfe/protocols/openmm_rfe/equil_rfe_settings.py]: done

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py
Comment (IAlibay) [reply openfe/protocols/openmm_rfe/equil_rfe_settings.py]: done on both counts

**Review by richardjgowers — APPROVED** [review]

**Comment by IAlibay** [comment]
To update on the current status of testing for this:

@hannahbaumann is leading the validation for this, she is looking into some of the validation tests that were done here: https://pubs.acs.org/doi/abs/10.1021/acs.jctc.8b00825

**Comment by richardjgowers** [comment]
@IAlibay can we merge this as a beta version, then potentially patch it?  I'd rather that than have to keep rebasing merge conflicts around it

**Comment by IAlibay** [comment]
> @IAlibay can we merge this as a beta version, then potentially patch it? I'd rather that than have to keep rebasing merge conflicts around it

I assume that's fine, but would defer to @hannahbaumann here - do you have any thoughts on this?

**Comment by hannahbaumann** [comment]
Sounds good to me! We may want to change the default number of lambda windows for charge changing perturbations before other people are using it, but that can also happen after the merge.

**Comment by IAlibay** [comment]
> Sounds good to me! We may want to change the default number of lambda windows for charge changing perturbations before other people are using it, but that can also happen after the merge.

I've raised #609 so we can discuss this - probably a separate PR would be better.

**Comment by IAlibay** [comment]
@richardjgowers as-is the correction is applied by default, if it's a beta feature would you agree to setting https://github.com/OpenFreeEnergy/openfe/pull/569/files#diff-74b77d39f244b9ced3853d19bef1c2f9b4ba5ba65064af79ef5daa2a188d1bd9R81 to `False`?

**Comment by IAlibay** [comment]
I've made the change for now to see how much of CI is going to break..

**Pull request merged**
